configlib/config.go

```diff
@@ -38,9 +38,61 @@ func NewConfig(cfg *PkgrConfig) {
 		rVersion := rcmd.GetRVersion(&rs)
 		cfg.Library = getLibraryPath(cfg.Lockfile.Type, cfg.RPath, rVersion, rs.Platform, cfg.Library)
 	}
+
+	// For all cfg values that can be paths, make sure that ~ is expanded to the home directory.
+	cfg.Library = expandTilde(cfg.Library)
+	cfg.RPath = expandTilde(cfg.RPath)
+	cfg.Tarballs = expandTildes(cfg.Tarballs)
+	cfg.Repos = expandTildesRepos(cfg.Repos)
+	cfg.Logging.All = expandTilde(cfg.Logging.All)
+	cfg.Logging.Install = expandTilde(cfg.Logging.Install)
+	cfg.Cache = expandTilde(cfg.Cache)
+
 	return
 }
 
+/// expand the ~ at the beginning of a path to the home directory.
+/// consider any problems a fatal error.
+func expandTilde(p string) string {
+	expanded, err := homedir.Expand(p)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"path": p,
+			"error": err,
+		}).Fatal("problem parsing config file -- could not expand path")
+	}
+	return expanded
+}
+
+/// For a list of paths, expand the ~ at the beginning of each path to the home directory.
+/// consider any problems a fatal error.
+func expandTildes(paths []string) []string {
+	var expanded []string
+	for _, p := range paths {
+		newPath := expandTilde(p)
+		expanded = append(expanded, newPath)
+	}
+	return expanded
+}
+
+/// In the PkgrConfig object, Repos are stored as a list of key-value pairs.
+/// Keys are repo names and values are paths to those repos
+/// For each key-value pair, expand the prefix ~ to be the home directory, if applicable.
+/// consider any problems a fatal error.
+func expandTildesRepos(repos []map[string]string) []map[string]string {
```

Comment:
I don't see any unit tests for this? Can you provide one? Or point to where it is?

Reply:
Primarily because it's the most complicated logic change to the branch

Reply:
Thanks for the review! 

I had intentionally omitted unit tests for this change because we're testing home directories via `~`, which aren't consistent across machines. I suspect this might be handleable via afero, but I haven't looked into it yet.

Reply:
well one simple test can be, for example, to set a repo given ~/blah, then expect that the return string does not contain ~/, or as DJ mentioned above, use one of the other mechanisms to derive what homedir.Expand _should_ do. 

something like:

```
userDir := os.UserHomeDir() // this is made up
assert.equal(repos[toExpand], userDir + 'hard/coded/rest')
```